scripts/build_test.py

```diff
@@ -101,3 +366,95 @@ def test_get_hashes_json_file_contents(self):
+        """Test build_files to queue up correct number of build tasks."""
+        build_tasks = collections.deque()
+        asset_hashes = build.get_file_hashes(build.ASSETS_DEV_DIR)
+        # Build all files.
+        build.build_files(
+            build.ASSETS_DEV_DIR, build.ASSETS_OUT_DIR, asset_hashes,
+            build_tasks)
+        total_file_count = build.get_file_count(build.ASSETS_DEV_DIR)
+        # Minus 1 to adjust for hashes.js.
+        self.assertEqual(total_file_count - 1, len(build_tasks))
+        build_tasks.clear()
+
+        # Only build HTML files.
+        build.build_files(
+            build.ASSETS_DEV_DIR, build.ASSETS_OUT_DIR, asset_hashes,
+            build_tasks, file_formats=('.html',))
+        total_html_file_count = 0
+        for _, _, files in os.walk(build.ASSETS_DEV_DIR):
+            for filename in files:
+                if filename.endswith('.html'):
+                    total_html_file_count += 1
+        self.assertEqual(total_html_file_count, len(build_tasks))
+
+    def test_rebuild_new_files(self):
+        """Test rebuid_new_files queue up a corresponding number of build tasks
+        to the number of file changes.
+        """
+        build_tasks = collections.deque()
+        new_file_name = 'manifest.json'
+        recently_changed_filenames = [
+            os.path.join(build.ASSETS_DEV_DIR, 'i18n', new_file_name)]
+        asset_hashes = build.get_file_hashes(build.ASSETS_DEV_DIR)
+        build.rebuild_new_files(
+            build.ASSETS_DEV_DIR, build.ASSETS_OUT_DIR,
+            recently_changed_filenames, asset_hashes, build_tasks)
+        self.assertEqual(len(build_tasks), len(recently_changed_filenames))
+
+    def test_get_recently_changed_filenames(self):
+        """Test get_recently_changed_filenames to detect file recently added."""
+        # Obtain hashes from DEV's /assets.
+        assets_hashes = build.get_file_hashes(build.ASSETS_DEV_DIR)
+        # Create an empty sandbox folder, to simulate an empty /build folder.
+        SANDBOX_DIR = os.path.join(build.PARENT_DIR, 'sandbox')
+        build.ensure_directory_exists(SANDBOX_DIR)
+        recently_changed_filenames = []
+        # Assert that list is init to 0.
+        self.assertEqual(len(recently_changed_filenames), 0)
+        recently_changed_filenames = build.get_recently_changed_filenames(
+            assets_hashes, SANDBOX_DIR)
+        # Since all HTML and Python files are already built, they are ignored.
+        with self.swap(build, 'FILE_EXTENSIONS_TO_IGNORE', ('.html', '.py',)):
+            self.assertEqual(
+                len(recently_changed_filenames), build.get_file_count(
+                    build.ASSETS_DEV_DIR) - 1)
+        # Minus 1 from ASSETS_DEV_DIR to account for added hashes.js.
+
+    def test_remove_deleted_files(self):
+        """Test remove_deleted_files to queue up the correct number of deletion
+        task.
+        """
+        delete_tasks = collections.deque()
```

Comment:
In general, let's structure tests like this (including newlines):

```
    {{Setup code}}

    {{Assertion of initial conditions}}
    {{Action being tested}}
    {{Assertion of final conditions}}

    {{Assertion of initial conditions 2}}
    {{Action being tested 2}}
    {{Assertion of final conditions 2}}

    ...
```

Reply:
So, here, it would be:

    delete_tasks = collections.deque()
    # The empty dict means that all files should be removed.
    file_hashes = dict()

    self.assertEqual(len(delete_tasks), 0)
    build.remove_deleted_files(file_hashes, ..., delete_tasks)
    self.assertEqual(len(delete_tasks), ...)

Reply:
Removed extra comments and added newline. Thanks!